client/blocks/inline-help/popover.jsx

```diff
@@ -32,46 +35,66 @@ class InlineHelpPopover extends Component {
 	};
 
 	state = {
-		showContactForm: false,
+		showSecondaryView: false,
+		activeSecondaryView: '',
 	};
 
-	openResult = ( event, href ) => {
-		if ( ! href ) {
-			return;
-		}
-
-		this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', {
-			search_query: this.props.searchQuery,
-			result_url: href,
-		} );
-
-		if ( ! event.metaKey ) {
-			event.preventDefault();
-			window.location = href;
-		} else if ( event.key === 'Enter' ) {
-			event.preventDefault();
-			window.open( href, '_blank' );
-		}
+	openResultView = event => {
+		event.preventDefault();
+		this.openSecondaryView( VIEW_RICH_RESULT );
 	};
 
 	moreHelpClicked = () => {
 		this.props.onClose();
 		this.props.recordTracksEvent( 'calypso_inlinehelp_morehelp_click' );
 	};
 
-	toggleContactForm = () => {
-		if ( this.state.showContactForm ) {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_contact_hide' );
-		} else {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_contact_show' );
-		}
-		this.setState( { showContactForm: ! this.state.showContactForm } );
+	setSecondaryViewKey = secondaryViewKey => {
+		this.setState( { activeSecondaryView: secondaryViewKey } );
+	};
+
+	openSecondaryView = secondaryViewKey => {
+		this.setSecondaryViewKey( secondaryViewKey );
+		this.props.recordTracksEvent( `calypso_inlinehelp_${ secondaryViewKey }_show` );
+		this.setState( { showSecondaryView: true } );
+	};
+
+	closeSecondaryView = secondaryViewKey => {
+		this.setSecondaryViewKey( '' );
```

Comment:
Is there benefit in keeping the `activeSecondaryView` in state even after we close (or 'hide') the view?
As in, if we had a general button that were to open the 'last secondary view' we would have lost that piece of data here so couldn't infer the 'last view key' by looking at `this.state.activeSecondaryView`.

If we're unlikely to a have a 'open last secondary view' type button then there's no harm in clearing this piece of state so isn't a blocker.

Reply:
> There are classes and conditional components that depend on the value being cleared. 

Would you mind pointing me to those?

Just because they've been written to depend on the value being cleared doesn't necessarily mean it's correct I'm not necessarily saying it's incorrect either - just curious.

Another thought: With `activeSecondaryView` as an empty string, are we likely to have things happen like setting a class like so: `.inline-help__` (no suffix) or tracking like so: `recordTracksEvent( 'calypso_inlinehelp__hide' );`  from `this.props.recordTracksEvent( 'calypso_inlinehelp_${ this.state.activeSecondaryView }_hide' );`?